src/server/cray/cfs/api/controllers/components.py

```diff
@@ -218,7 +218,7 @@ def patch_v2_components_dict(data):
         for component_id in id_list:
             component_data = DB.get(component_id)
             if component_data:
-                components.append(component_data)
+                components.append((component_id, component_data))
```

Comment:
What does DB.get_all() return? Does it return the same tuple of (component_id, component_data). I couldn't tell from looking at it. It returns a component_data page, whatever that is, but if it's not the same as the tuple above, then you're going to have problems when the id_list is empty.

Reply:
https://github.com/Cray-HPE/config-framework-service/pull/119